### ISSUE TITLE
[Feature Request] Allow items to traverse DimDungeons portals

### ISSUE THREAD

**Issue by Ronn0k** [AllenSeitz/DimDungeons#83]
Hi

The dungeons and personal dimension are awesome. Love it. 8x8 chunks seems like a reasonable amount of space and is pefect imo. 

However, I've built a large base within the personal dimension with many mods. Unfortunately I've hit a roadblock where I cannot get items or fluids in/out of the personal dimension in any automated manner.

This prevents many things, but to use my current issue as an example, I need to get oil to create fuel. I have no way of getting oil into the dimension other than doing it manually myself. This essentially means I need to build an entire power generation system outside the dimension, as well as any and everything that requires oil or fuel; otherwise it would require manual back and forth on an unfun scale.

The usual solution when a mod/modpack does not have interdimensional item/fluid/power transfer devices, is to have blocks 'fall' into the portal, eg the nether portal, and be collected on the other side via hoppers or similar things. Fluid/power can usually be transferred this way also via buckets or batteries.

Is this a possible feature that may be implemented in the future? 
As it is currently, using the personal dimension for a base is a very bad idea if anything like what I described above is required, and there other ways to move things easily between dimensions with pipes, tanks, chests etc.

Thanks!
Ronn0k
(Currently playing Create Chronicles: Bosses and Beyond)

**Comment by AllenSeitz** [comment]
Good idea. It was easy to make the dungeon portal player-only. This solved a number of problems. For one thing I didn't have to worry about Creepers or the Ender Dragon randomly flying into the portal and getting teleported inside the dungeon. Certain vehicles like boats also caused strange issues,

But if the entity is an ItemStack then I don't see a problem. Maybe even, if the entity is an ItemStack and the destination is the personal dimension, then allow it? Because maybe it might be possible to exploit throwing ItemStacks into or out of a dungeon if the hardcore option is set in the server config. 

I'm still very deep into my 1.21 port. I'll probably implement this for 1.21 and I'll try to back port it to 1.20.1. Would that help with your modpack? (Assuming the modpack chose to update.)

**Comment by Ronn0k** [comment]
Thanks for the response. Damn that was quick.
I see why the reasoning behind making the portal player-only.

That suggestion would definitely solve the issue. Given the portal is 1x2 however, having items dropped or pushed into it by various machines/pipes/hoppers would clog the space a bit, but that is simply an inconvenience and minor compared to it not being possible.

Using a teleporter room as the main entrance would solve that issue, leaving the main portal for moving items etc, however the main portal is always on the edge of the 8x8. Small QoL things to think about if you do implement it, and they're not too time consuming.

Regardless of the outcome of this request, know that your mod and all the work and effort is appreciated!!
